linux/LinuxMachine.c

```diff
@@ -78,7 +78,7 @@ static void LinuxMachine_updateCPUcount(LinuxMachine* this) {
          continue;
 
 #ifdef HAVE_OPENAT
-      int cpuDirFd = openat(dirfd(dir), entry->d_name, O_DIRECTORY | O_PATH | O_NOFOLLOW);
+      int cpuDirFd = openat(xDirfd(dir), entry->d_name, O_DIRECTORY | O_PATH | O_NOFOLLOW);
```

Comment:
How is this `xDirfd` able to silence the invalid file descriptor warning? I don't get it. Since the assertions can be turned off, the warning should exist if you are not building in the debug mode.

Reply:
It's just to please the static analyzer for development and in the CI. Since Linux does support file descriptors for directories `dirfd()` never fails.